### ISSUE TITLE
Test failure: websocket.client/ClientCloseTest.testStopLifecycle

### ISSUE THREAD

**Issue by sbordet** [eclipse/jetty.project#2625]
https://jenkins.webtide.net/job/jetty.project/job/jetty-9.4.x/389/testReport/junit/org.eclipse.jetty.websocket.client/ClientCloseTest/linux_jdk10___Test___jdk10___testStopLifecycle/

**Comment by joakime** [comment]
Details. (JDK 10 build only)

```
Error
test timed out after 5000 milliseconds

Stacktrace
org.junit.runners.model.TestTimedOutException: test timed out after 5000 milliseconds
at app//org.eclipse.jetty.websocket.client.ClientCloseTest.testStopLifecycle(ClientCloseTest.java:573)

Standard Error
2018-06-05 17:11:44.319:WARN:oejut.QueuedThreadPool:main: QueuedThreadPool[HttpClient@380e33c2]@29188050{STOPPING,8<=8<=200,i=0,q=9}[org.eclipse.jetty.util.thread.TryExecutor$$Lambda$71/1778801068@492c8137] Couldn't stop Thread[HttpClient@380e33c2-468,5,main]
```

**Comment by joakime** [comment]
This test was reviewed as part of the work on #2655 and has been re-enabled without incident for a week.
Please reopen if it occurs again.

**Comment by lachlan-roberts** [comment]
I am getting intermittent test failures for this test in jetty-10.0.x
seems to happen about 20% of the times I run the test

```
2019-03-21 15:32:35.472:INFO::main: Logging initialized @1212ms to org.eclipse.jetty.util.log.StdErrLog
2019-03-21 15:32:37.114:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-25: onWebSocketConnect(WebSocketSession[CLIENT,to=PT1S,WSChannel@2de50278{CLIENT,WebSocketChannelState@85ce376{CONNECTED,i=NO-OP,o=NO-OP,c=null},[ws://192.168.1.9:61178/ws,null,false.[]],af=true,i/o=4096/4096,fs=65536}->JettyWebSocketFrameHandler@42dc5b00[org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint],JettyWebSocketFrameHandler@42dc5b00[org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint]])
2019-03-21 15:32:37.127:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-21: onWebSocketText(echo-test)
2019-03-21 15:32:37.151:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-20: onWebSocketConnect(WebSocketSession[CLIENT,to=PT1S,WSChannel@383f178a{CLIENT,WebSocketChannelState@7d3d8ee9{CONNECTED,i=NO-OP,o=NO-OP,c=null},[ws://192.168.1.9:61178/ws,null,false.[]],af=true,i/o=4096/4096,fs=65536}->JettyWebSocketFrameHandler@74141d18[org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint],JettyWebSocketFrameHandler@74141d18[org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint]])
2019-03-21 15:32:37.161:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-27: onWebSocketText(echo-test)
2019-03-21 15:32:37.169:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-24: onWebSocketConnect(WebSocketSession[CLIENT,to=PT1S,WSChannel@f1442e5{CLIENT,WebSocketChannelState@4a16cd63{CONNECTED,i=NO-OP,o=NO-OP,c=null},[ws://192.168.1.9:61178/ws,null,false.[]],af=true,i/o=4096/4096,fs=65536}->JettyWebSocketFrameHandler@2e52e535[org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint],JettyWebSocketFrameHandler@2e52e535[org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint]])
2019-03-21 15:32:37.171:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-27: onWebSocketText(echo-test)
2019-03-21 15:32:37.185:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-20: onWebSocketError
java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.websocket.core.internal.WebSocketChannel.onEof(WebSocketChannel.java:310)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onClose(WebSocketConnection.java:174)
	at org.eclipse.jetty.io.SelectorManager.connectionClosed(SelectorManager.java:326)
	at org.eclipse.jetty.io.ManagedSelector$DestroyEndPoint.run(ManagedSelector.java:960)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.base/java.lang.Thread.run(Thread.java:834)
2019-03-21 15:32:37.185:DBUG:oejwtc.ClientCloseTestServerEndpoint:qtp2073707154-16: 
java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.websocket.core.internal.WebSocketChannel.onEof(WebSocketChannel.java:310)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.fillAndParse(WebSocketConnection.java:435)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onFillable(WebSocketConnection.java:292)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:309)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:365)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.base/java.lang.Thread.run(Thread.java:834)
2019-03-21 15:32:37.185:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-22: onWebSocketError
java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.websocket.core.internal.WebSocketChannel.onEof(WebSocketChannel.java:310)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onClose(WebSocketConnection.java:174)
	at org.eclipse.jetty.io.SelectorManager.connectionClosed(SelectorManager.java:326)
	at org.eclipse.jetty.io.ManagedSelector$DestroyEndPoint.run(ManagedSelector.java:960)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.base/java.lang.Thread.run(Thread.java:834)
2019-03-21 15:32:37.187:DBUG:oejwtc.ClientCloseTestServerEndpoint:qtp2073707154-18: 
java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.websocket.core.internal.WebSocketChannel.onEof(WebSocketChannel.java:310)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.fillAndParse(WebSocketConnection.java:435)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onFillable(WebSocketConnection.java:292)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:309)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:365)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.base/java.lang.Thread.run(Thread.java:834)
2019-03-21 15:32:37.185:DBUG:oejwtc.ClientCloseTestServerEndpoint:qtp2073707154-19: 
java.nio.channels.ClosedChannelException
	at org.eclipse.jetty.websocket.core.internal.WebSocketChannel.onEof(WebSocketChannel.java:310)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.fillAndParse(WebSocketConnection.java:435)
	at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onFillable(WebSocketConnection.java:292)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:309)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:365)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.base/java.lang.Thread.run(Thread.java:834)
2019-03-21 15:32:37.190:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-20: onWebSocketClose(1006,Channel Closed)
2019-03-21 15:32:37.191:DBUG:oejwt.CloseTrackingEndpoint:WebSocketClient@702846463-22: onWebSocketClose(1006,Channel Closed)

java.lang.AssertionError: Client Close Event Occurred
Expected: is <true>
     but: was <false>
Expected :is <true>
     
Actual   :<false>
 <Click to see difference>


	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.eclipse.jetty.websocket.tests.CloseTrackingEndpoint.assertReceivedCloseEvent(CloseTrackingEndpoint.java:65)
	at org.eclipse.jetty.websocket.tests.client.ClientCloseTest.testStopLifecycle(ClientCloseTest.java:332)
```

**Comment by lachlan-roberts** [comment]
test now fixed in jetty-10 as part of the changes for #3494
